rest-api/test/unit_test/api_test/public_metrics_api_test.py

```diff
@@ -0,0 +1,923 @@
+
+    if first_name:
+      summary.firstName = first_name
+    if last_name:
+      summary.lastName = last_name
+
+    if gender_id:
+      summary.genderIdentityId = gender_id
+    if dob:
+      summary.dateOfBirth = dob
+    else:
+      summary.dateOfBirth = datetime.date(1978, 10, 10)
+    if state_id:
+      summary.stateId = state_id
+
+    if primary_language:
+      summary.primaryLanguage = primary_language
+
+    summary.enrollmentStatus = enrollment_status
+
+    summary.enrollmentStatusMemberTime = time_mem
+    summary.enrollmentStatusCoreOrderedSampleTime = time_fp
+    summary.enrollmentStatusCoreStoredSampleTime = time_fp_stored
+
+    summary.hpoId = self.hpo_dao.get_by_name(hpo_name).hpoId
+
+    if time_study is not None:
+      with FakeClock(time_mem):
+        summary.consentForStudyEnrollmentTime = time_study
+
+    if time_mem is not None:
+      with FakeClock(time_mem):
+        summary.consentForElectronicHealthRecords = 1
+        summary.consentForElectronicHealthRecordsTime = time_mem
+
+    if time_fp is not None:
+      with FakeClock(time_fp):
+        if not summary.consentForElectronicHealthRecords:
+          summary.consentForElectronicHealthRecords = 1
+          summary.consentForElectronicHealthRecordsTime = time_fp
+        summary.questionnaireOnTheBasicsTime = time_fp
+        summary.questionnaireOnLifestyleTime = time_fp
+        summary.questionnaireOnOverallHealthTime = time_fp
+        summary.questionnaireOnHealthcareAccessTime = time_fp
+        summary.questionnaireOnMedicalHistoryTime = time_fp
+        summary.questionnaireOnMedicationsTime = time_fp
+        summary.questionnaireOnFamilyHealthTime = time_fp
+        summary.physicalMeasurementsFinalizedTime = time_fp
+        summary.physicalMeasurementsTime = time_fp
+        summary.sampleOrderStatus1ED04Time = time_fp
+        summary.sampleOrderStatus1SALTime = time_fp
+        summary.sampleStatus1ED04Time = time_fp
+        summary.sampleStatus1SALTime = time_fp
+
+    self.ps_dao.insert(summary)
+
+    return summary
+
+  def test_public_metrics_get_enrollment_status_api(self):
+
+    p1 = Participant(participantId=1, biobankId=4)
+    self._insert(p1, 'Alice', 'Aardvark', 'UNSET', unconsented=True, time_int=self.time1)
+
+    p2 = Participant(participantId=2, biobankId=5)
+    self._insert(p2, 'Bob', 'Builder', 'AZ_TUCSON', time_int=self.time2)
+
+    p3 = Participant(participantId=3, biobankId=6)
+    self._insert(p3, 'Chad', 'Caterpillar', 'AZ_TUCSON', time_int=self.time1, time_mem=self.time3,
+                 time_fp_stored=self.time4)
+
+    service = ParticipantCountsOverTimeService()
+    dao = MetricsEnrollmentStatusCacheDao(MetricsCacheType.PUBLIC_METRICS_EXPORT_API)
+    service.refresh_data_for_metrics_cache(dao)
+
+    qs = """
+          &stratification=ENROLLMENT_STATUS
+          &startDate=2018-01-01
+          &endDate=2018-01-08
+          """
+
+    qs = ''.join(qs.split())  # Remove all whitespace
```

Comment:
There is nothing wrong with this way of cleaning whitespace out of a string. There is a more common pythonic way though:
```
qs = (
  '&stratification=ENROLLMENT_STATUS'
  '&startDate=2018-01-01'
  '&endDate=2018-01-08'
)
```

Python automatically concatenates consecutive string literals.

Reply:
Cool, thanks for the tip! I will change it to this way.